src/lib/services/blockchainWatcher.ts

```diff
@@ -1,23 +1,27 @@
 // a temporary placeholder for the blockchain watcher
 
 import { UUID } from '../../models/uuid'
-import { TokenType } from '../../models/tokenType'
+import { TOKEN_TYPE } from '../../models/tokenType'
 import Database from '../db'
 import { DemandState, Match2State } from '../../models'
 
 const db = new Database()
 
-const typeTableMap = {
-  [TokenType.DEMAND]: 'demand',
-  [TokenType.MATCH2]: 'match2',
+const typeToTable = (tokenType: TOKEN_TYPE) => {
+  switch (tokenType) {
+    case 'DEMAND':
+      return 'demand'
+    case 'MATCH2':
+      return 'match2'
+  }
```

Comment:
const typeToTable = (tokenType: TOKEN_TYPE) => tokenType.toLowerCase()? why switch? Just out of interest

Reply:
it forces the correct type coertion. I tried to do this with an object map but typescript wasn't happy with that